paywall/src/components/content/CheckoutContent.tsx

```diff
@@ -61,17 +62,24 @@ export default function CheckoutContent() {
     account,
     network,
     locks,
+    checkWallet,
   }: blockchainData = (useBlockchainData as useBlockchainDataFunc)(
     window,
     paywallConfig
   )
 
+  const [showWalletCheckOverlay, setShowWalletCheckOverlay] = useState(false)
+  const [walletOverlayDismissed, setWalletOverlayDismissed] = useState(false)
```

Comment:
It's kind of gross but we want to only show the modal when `checkWallet` is set to true and the user has not explicitly dismissed it. After that, they're on their own.